osu.Game/Overlays/NewsOverlay.cs

```diff
@@ -49,39 +53,62 @@ private void load(OsuColour colours)
                             {
                                 ShowFrontPage = ShowFrontPage
                             },
-                            content = new Container<NewsContent>
+                            content = new Container
                             {
                                 RelativeSizeAxes = Axes.X,
                                 AutoSizeAxes = Axes.Y,
                             }
                         },
                     },
                 },
+                loading = new LoadingLayer(content),
             };
 
             header.Post.BindTo(Current);
-            Current.TriggerChange();
         }
 
-        private CancellationTokenSource loadContentCancellation;
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            Current.BindValueChanged(onCurrentChanged, true);
+        }
+
+        public void ShowFrontPage()
+        {
+            Current.Value = null;
+            Show();
+        }
+
+        private CancellationTokenSource cancellationToken;
 
-        protected void LoadAndShowContent(NewsContent newContent)
+        private void onCurrentChanged(ValueChangedEvent<string> current)
         {
-            content.FadeTo(0.2f, 300, Easing.OutQuint);
+            cancellationToken?.Cancel();
+            loading.Show();
+
+            if (current.NewValue == null)
+            {
+                LoadDisplay(new FrontpageDisplay());
+                return;
+            }
 
-            loadContentCancellation?.Cancel();
+            LoadDisplay(Empty());
+        }
```

Comment:
So this stuff was here before, more or less, but I dislike it. Displaying the front page by setting a `null` value to a `string` bindable is not acceptable in my view (and why are both `ShowFrontPage()` and `Current` public?) I'd rather just have actual methods like `ShowFrontPage()` and `ShowArticle(string slug)` or something.